TextFieldEffects/TextFieldEffects/TextFieldEffects.swift

```diff
@@ -100,6 +106,12 @@ open class TextFieldEffects : UITextField {
         }
     }
     
+    open override func sizeThatFits(_ size: CGSize) -> CGSize {
+        let textFieldSize = super.sizeThatFits(size)
+        let placeholderSize = placeholderLabel.sizeThatFits(size)
+        return CGSize(width: max(textFieldSize.width, placeholderSize.width), height: textFieldSize.height + placeholderSize.height)
```

Comment:
This will break the project. If you are having problems with your frames, make sure your constraints are setup correctly, but its also important to know that the project also does a lot of "math on CGRect" and there is always a minimum height that needs to be present in order for some of the fields to work.

They won't work for "any size"